### PR TITLE
Use upstream aws base hook session implementation 

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/astronomer/astronomer-providers/issues/1227

We have missing implementation for some auth type so I was thinking if it makes sense to use upstream provider base hook implementation. 

Screenshot 2023-07-05 at 12 40 33 AM

### CODE REVIEW

**Comment by pankajastro** [comment]
@phanikumv wdyt?

**Comment by phanikumv** [comment]
> @phanikumv wdyt?

yes it makes sense

**Review by pankajkoti — COMMENTED** [review]
Let's test all our example DAGs on Astro before merging these.

File: astronomer/providers/amazon/aws/hooks/base_aws.py [hunk 1]
```diff
@@ -34,6 +34,12 @@ class AwsBaseHookAsync(AwsBaseHook):
 
     async def get_client_async(self) -> AioBaseClient:
         """Create an Async Client object to communicate with AWS services."""
+        # Try to use airflow provider client implementation
+        try:
+            return self._get_aioclient()
```
Comment (phanikumv) [inline astronomer/providers/amazon/aws/hooks/base_aws.py:39]: Let's also include assume role config in our aws connection so that we can test it

Comment (pankajastro) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:39]: I added some  https://github.com/apache/airflow/pull/28850/files#diff-4e7fd0a0e31b60e534f7cab71ee4d3591c0758c1e76054f2403b6669546cb2d2 do you want me to replicate the same here

Comment (phanikumv) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:39]: I was looking for an integration test for `assume_role` functionality

Comment (pankajastro) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:39]: created connection in cloud account and added a operator please take a look

File: astronomer/providers/amazon/aws/example_dags/example_batch.py [hunk 2]
```diff
@@ -287,6 +288,16 @@ def check_dag_status(**kwargs: Any) -> None:
     )
     # [END howto_operator_batch_async]
 
+    submit_batch_job_assume_cred = BatchOperatorAsync(
+        task_id="assume_role_cred",
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/example_dags/example_batch.py:292]:
```suggestion
        task_id="submit_job_with_assume_role",
```

File: astronomer/providers/amazon/aws/example_dags/example_batch.py [hunk 3]
```diff
@@ -287,6 +288,16 @@ def check_dag_status(**kwargs: Any) -> None:
     )
     # [END howto_operator_batch_async]
 
+    submit_batch_job_assume_cred = BatchOperatorAsync(
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/example_dags/example_batch.py:291]:
```suggestion
    submit_job_with_assume_role = BatchOperatorAsync(
```

File: astronomer/providers/amazon/aws/example_dags/example_batch.py [hunk 4]
```diff
@@ -336,6 +347,7 @@ def check_dag_status(**kwargs: Any) -> None:
         >> submit_batch_job
         >> list_jobs
         >> batch_job_sensor
+        >> submit_batch_job_assume_cred
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/example_dags/example_batch.py:350]:
```suggestion
        >> submit_job_with_assume_role
```

File: astronomer/providers/amazon/aws/hooks/base_aws.py (hunk 1 above)
Comment (phanikumv) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:43]: @pankajastro Could not find it on our deployments. Can you please let me know where this is added

File: astronomer/providers/amazon/aws/hooks/base_aws.py [hunk 5]
```diff
@@ -152,3 +158,12 @@ async def get_role_credentials(
                 )
                 return_response = response["Credentials"]
             return return_response
+
+    def _get_aioclient(self) -> AioBaseClient:
+        session = self.get_session(region_name=self.region_name, deferrable=True)
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/hooks/base_aws.py:163]: Would this require bumping our minimum amazon provider version? I think the deferrable flag would have been added recently right?

File: astronomer/providers/amazon/aws/hooks/base_aws.py (hunk 1 above)
Comment (pankajastro) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:43]: It was https://org-airflow-team-org.astronomer.run/dfi0rona/connection/list/

File: astronomer/providers/amazon/aws/hooks/base_aws.py (hunk 5 above)
Comment (pankajkoti) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:163]: Not sure if it has all the needed pieces, I just extracted custom pieces and tailored in that PR to add support for refreshable credentials for only the `assume_role` method

Comment (pankajastro) [reply astronomer/providers/amazon/aws/hooks/base_aws.py:163]: I think we should do a similar for other methods too which have been not covered

**Comment by pankajastro** [comment]
changing this to draft to avoid accidental merge

**Pull request closed without merging**